modules/common/src/main/scala/surge/core/SurgeFormatting.scala

```diff
@@ -22,6 +23,8 @@ trait SurgeEventWriteFormatting[Event] {
 trait SurgeAggregateWriteFormatting[Agg] {
   def writeState(agg: Agg): SerializedAggregate
 }
+
+@deprecated("Encourages the use of an anti-pattern. Use SurgeAggregateWriteFormatting and SurgeEventWriteFormatting separately.", "0.5.4")
```

Comment:
Should we also deprecate `SurgeAggregateFormatting` and `SurgeEventFormatting` for the same reason?

Reply:
No, I think it makes sense to have something that knows how to serde an aggregate or an event but not something that serializes aggregates and events and something else that deserializes them.